networkit/graphtools.pyx

```diff
@@ -705,3 +706,20 @@ cdef class GraphTools:
 		"""
 		result = createAugmentedGraph(G._this)
 		return Graph().setThis(result.first), result.second
+
+	@staticmethod
+	def randomizeWeights(Graph G):
+		"""
+		randomizeWeights(G)
+
+		Randomizes the weights of the given graph. The weights are uniformly distributed in
+		the range [0, 1] by default, unless a different distribution is provided. However it
+		is only strictly in-place for already weighted graphs. For unweighted graphs a copy is
+		created before randomizing weights.
+
+		Parameters
+		----------
+		G : networkit.Graph
+			The directed input graph.
+		"""
+		randomizeWeights(G._this)
```

Comment:
Please add some additional checks, e.g., that the graph is weighted and the edge weights are within the expected values.

Reply:
Why is this necessary? If the graph is unweighted then the graph will be made to be weighted and the previous edge weights are ignored anyway.

Reply:
Ah sorry, I thought that this was a test.